File: LeetcodeProblems/2053.py
# beats 85% in time and 82% in space

class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        F = Counter(arr)
        dist = list(filter(lambda x: F[x] == 1, arr))
        if len(dist) < k:
            return ""
        return dist[k - 1]

File: LeetcodeProblems Cpp/2053.cpp
class Solution {
public:
    string kthDistinct(vector<string>& arr, int k) {
        std::unordered_map<std::string, int> F;
        for (auto& el: arr) {
            F[el]++;
        }
        int count = 0;
        for (int i = 0; i < arr.size(); i++) {
            if (F[arr[i]] == 1) {
                count++;

            }
            if (count == k) {
                return arr[i];
            }
        }
        return "";
    }
};